content/blaze.md

```diff
@@ -0,0 +1,543 @@
+
+You can read about the full Spacebars syntax [here](https://github.com/meteor/meteor/blob/devel/packages/spacebars/README.md), but in this section we'll attempt to cover some important finer details.
+
+<h3 id="data-contexts">Data contexts and lookup</h3>
+We've seen `{{todo.title}}` accesses the `title` property of the `todo` item on the current data context. Additionaly, `..` access the parent data context (rarely a good idea), `list.todos.[0]` accesses the first element of the `todos` array on `list`.
+
+Also, note that Spacebars is very forgiving of `null` values. It will not complain if you try to access a property on a `null` value (for instance `foo.bar` if `foo` is not defined), but instead simply treat it also as null. However there are exceptions to this---trying to call a `null` function, or doing the same *within* a helper will lead to exceptions.
+
+<h3 id="helpers">Calling helpers with arguments</h3>
+You can provide arguments to a helper like `checkedClass` by simply placing the argument after the helper call, as in: `{{checkedClass todo true 'checked'}}`. You can also provide a list of named keyword arguments to a helper with `{{checkedClass todo noClass=true classname='checked'}}. In this case, you might access those arguments with:
+
+```js
+Template.todosItem.helpers({
+  checkedClass(todo, kws) {
+    const classname = kws.hash.classname || 'checked';
+    if (todo.checked) {
+      return classname;
+    } else if (kws.hash.noClass) {
+      return `no-${classname}`;
+    }
+  }
+});
+```
+
+You can also pass the output of a helper to a template inclusion or other helper. To do so, use brackets to show precedence:
+
+```html
+{{> todosItem (todoArgs todo)}}
+```
+
+Here the `todo` is passed as argument to the `todoArgs` helper, then the output is passed into the `todosItem` template.
+
+<h3 id="inclusion">Template inclusion</h3>
+You "include" a subtemplate with the `{{>` syntax. By default, the sub-template will gain the data context of the caller, although it's usually a good idea to be explicit. You can provide a single object as argument (as we did with the object returned by the `todoArgs` helper above), or provide a list of keyword arguments, as the `listShowPage` template does:
+
+```html
+{{> listsShow todosReady=Template.subscriptionsReady
+  list=(getFullList listIdOnly) todos=listIdOnly.todos}}
+```
+
+In this case, the `listShow` template can expect a data context of the form:
+
+```js
+{
+  todosReady: ...,
+  list: ...,
+  todos: ...
+}
+```
+
+<h3 id="helpers-in-tags">Helpers in tags</h3>
+We saw above that using a helper (or data context lookup) in the form `checked={{todo.checked}}` will add the checked property to the HTML tag if `todo.checked` evaluates to true. Also, you can directly include an object as part of an HTML to apply multiple properties at once:
```

Comment:
"include an object as part of an HTML to apply multiple properties at once:" -> "Include an object in the attribute list of an HTML element to set multiple attributes at once."